### PR TITLE
[cleanup]Enable "no-floating-promises" in notification-service package

### PR DESCRIPTION
**Pull request by ashishb**
### Description

Enable no-floating-promises in notification-service package

### Tested

`packages/notification-service $ yarn lint`
Can't think of any better tests here without re-deployment.

### CODE REVIEW

File: packages/notification-service/src/firebase.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ export function setLastBlockNotified(newBlock: number) {
   // we set it here ourselves to avoid race condition where we check for notifications
   // again before it syncs
   lastBlockNotified = newBlock
-  lastBlockRef.set(newBlock)
+  await lastBlockRef.set(newBlock)
```
Comment (jmrossy) [inline packages/notification-service/src/firebase.ts]: Ditto as above. Lets replace await with return here and make this function not async

File: packages/notification-service/src/blockscout/transfers.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ function notifyForNewTransfers(transfers: Transfer[], lastBlockNotified: number)
       blockNumber: String(t.blockNumber),
       timestamp: String(t.timestamp),
     }
-    sendPaymentNotification(
+    await sendPaymentNotification(
```
Comment (jmrossy) [inline packages/notification-service/src/blockscout/transfers.ts]: If we await here, we will send each notification serially.  The idea is to fire them all off asap.
one option would be to collect the promises in an array and then await for `Promise.all()`

File: packages/notification-service/src/blockscout/transfers.ts [hunk 3]
```diff
@@ -117,6 +117,6 @@ export async function handleTransferNotifications() {
 
   const allTransfers = filterAndJoinTransfers(goldTransfers, stableTransfers)
 
-  notifyForNewTransfers(allTransfers, lastBlockNotified)
-  setLastBlockNotified(Math.max(goldTransfersLatestBlock, stableTransfersLatestBlock))
+  await notifyForNewTransfers(allTransfers, lastBlockNotified)
+  await setLastBlockNotified(Math.max(goldTransfersLatestBlock, stableTransfersLatestBlock))
```
Comment (jmrossy) [inline packages/notification-service/src/blockscout/transfers.ts]: the last line of a function should never be await, or you're better off just returning the promise object itself.

File: packages/notification-service/src/firebase.ts [hunk 4]
```diff
@@ -130,11 +130,11 @@ export function getPendingRequests() {
   return pendingRequests
 }
 
-export function setPaymentRequestNotified(uid: string) {
-  database.ref(`/pendingRequests/${uid}`).update({ notified: true })
+export async function setPaymentRequestNotified(uid: string) {
+  await database.ref(`/pendingRequests/${uid}`).update({ notified: true })
```
Comment (jmrossy) [inline packages/notification-service/src/firebase.ts]: Ditto as above. Lets replace await with return here and make this function not async

**Review by jmrossy — COMMENTED** [review]
This does rightly point out that some promises weren't being fully covered but we need to make some tweaks to not lose performance :)

File: packages/notification-service/src/firebase.ts (hunk 1 above)
Comment (ashishb) [reply packages/notification-service/src/firebase.ts]: Done

File: packages/notification-service/src/firebase.ts (hunk 4 above)
Comment (ashishb) [reply packages/notification-service/src/firebase.ts]: Done

File: packages/notification-service/src/blockscout/transfers.ts (hunk 2 above)
Comment (ashishb) [reply packages/notification-service/src/blockscout/transfers.ts]: Changed to `Promise.all`

File: packages/notification-service/src/blockscout/transfers.ts (hunk 3 above)
Comment (ashishb) [reply packages/notification-service/src/blockscout/transfers.ts]: Changed to `return`

**Review by jmrossy — APPROVED** [review]
LGTM!

**Pull request merged**